### PR TITLE
chore(deps): update driver and libmongocrypt MONGOSH-1550

### PR DESCRIPTION
**Pull request by mabaasit**

### CODE REVIEW

File: packages/shell-api/src/helpers.ts [hunk 1]
```diff
@@ -937,9 +936,11 @@ export function processFLEOptions(
       );
     }
   } else {
+    // TODO: switch from any to KMSProviders (@mongosh/service-provider-core) once
+    // https://github.com/mongodb/node-mongodb-native/pull/4160/files is released.
     autoEncryption.kmsProviders = {
       ...fleOptions.kmsProviders,
-    } as KMSProviders;
+    } as any;
```
Comment (addaleax) [inline packages/shell-api/src/helpers.ts:943]:
```suggestion
    } as any as KMSProviders;
```

makes it a bit clearer what the intended type is?

File: packages/e2e-tests/test/e2e-fle.spec.ts [hunk 2]
```diff
@@ -716,11 +716,158 @@ describe('FLE tests', function () {
...
     });
 
-    context('using rangePreview algorithm', function () {
-      // TODO(MONGOSH-1742): Server 8.0 drops "rangePreview" algorithm and adds
-      // "range". Re-enable these when the change is finalized
-      skipIfServerVersion(testServer, '>= 8.0.0-alpha');
+    context(
+      'using rangePreview algorithm when server is > 7 and < 8',
+      function () {
+        skipIfServerVersion(testServer, '>= 8.0.0');
```
Comment (addaleax) [inline packages/e2e-tests/test/e2e-fle.spec.ts:722]: Could we use the same test code for 7.x and 8.x+ and only switch between `range` and `rangePreview` according to the server version?

Comment (mabaasit) [reply packages/e2e-tests/test/e2e-fle.spec.ts:722]: yeah good point, will do it!

**Comment by mabaasit** [comment]
Closing in favour of https://github.com/mongodb-js/mongosh/pull/2076

**Pull request closed without merging**